src/lib/mina/transaction.ts

```diff
@@ -443,21 +446,21 @@ function transaction(f: () => void): Promise<Transaction>;
  */
 function transaction(
   sender: DeprecatedFeePayerSpec,
-  f: () => void
+  f: () => Promise<void>
 ): Promise<Transaction>;
 function transaction(
-  senderOrF: DeprecatedFeePayerSpec | (() => void),
-  fOrUndefined?: () => void
+  senderOrF: DeprecatedFeePayerSpec | (() => Promise<void>),
```

Comment:
Might be about time that we remove the old deprecated private key as fee payer spec

Reply:
good idea. I'll do an API cleanup PR before v1 that removes all the long deprecated stuff